pkg/openstack/loadbalancer_test.go

```diff
@@ -1543,3 +1543,99 @@ func Test_nodeAddressForLB(t *testing.T) {
 		})
 	}
 }
+
+func TestLbaasV2_getNetworkID(t *testing.T) {
+	lbaas := &LbaasV2{
+		LoadBalancer: LoadBalancer{
+			opts: LoadBalancerOpts{
+				LBClasses: map[string]*LBClass{
+					"lbclassKey": {
+						NetworkID: "lb-class-network-id-1234",
+					},
+				},
+				NetworkID: "default-lb-class-networkId",
+			},
+		},
+	}
+
+	type testArg struct {
+		service *corev1.Service
+	}
+	tests := []struct {
+		name    string
+		lbaas   LbaasV2
+		arg     testArg
+		want    string
+		wantErr string
+	}{
+		{
+			name:  "get network id from service annotation",
+			lbaas: LbaasV2{},
+			arg: testArg{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerNetworkID: "subnet-id",
+						},
+					},
+				},
+			},
+			want:    "subnet-id",
+			wantErr: "",
+		},
+		{
+			name:  "get network id from config class",
+			lbaas: *lbaas,
+			arg: testArg{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerClass: "lbclassKey",
+						},
+					},
+				},
+			},
+			want:    "lb-class-network-id-1234",
+			wantErr: "",
+		},
+		{
+			name:  "get network id from default config",
+			lbaas: *lbaas,
+			arg: testArg{
+				service: &corev1.Service{},
+			},
+			want:    "default-lb-class-networkId",
+			wantErr: "",
+		},
+		{
+			name:  "error when loadbalancer class not found",
+			lbaas: LbaasV2{},
+			arg: testArg{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerClass: "invalid-lb-class",
+						},
+					},
+				},
+			},
+			want:    "",
+			wantErr: "invalid loadbalancer class \"invalid-lb-class\"",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			lbaas := tt.lbaas
+
+			got, err := lbaas.getNetworkID(tt.arg.service, &serviceConfig{})
+
+			if tt.wantErr != "" {
+				assert.EqualError(t, err, tt.wantErr)
+			} else {
+				if assert.NoError(t, err) {
+					assert.Equal(t, tt.want, got)
+				}
+			}
```

Comment:
Why is there a nested `if`?

In this `else` branch, `tt.wantErr` is the empty string. This means that we expect `err` to be nil, full stop.

```suggestion
			} else {
				assert.NoError(t, err)
			}

Reply:
![image](https://github.com/kubernetes/cloud-provider-openstack/assets/54990092/2f1d35d9-99bc-424b-97c1-8b34ed3a377e)

i tried following the suggested principle that assert if the got and expected are equal if there is no error :)

i made the fixed already and sent them in @pierreprinetti

Reply:
ah! So that is where this pattern is coming from, thanks.

Reply:
I don't like it.

Reply:
yeah, i fixed it already :-)

Reply:
apparently, i'm learning alot in the process 
thank you very much